model/src/pyrenew/latent/infectionswithfeedback.py

```diff
@@ -0,0 +1,203 @@
+    is the max-length of the infection feedback pmf.
+    """
+
+    def __init__(
+        self,
+        infection_feedback_strength: RandomVariable,
+        infection_feedback_pmf: RandomVariable,
+        infections_mean_varname: str = "latent_infections",
+    ) -> None:
+        """
+        Default constructor for Infections class.
+
+        Parameters
+        ----------
+        infection_feedback_strength : RandomVariable
+            Infection feedback strength.
+        infection_feedback_pmf : RandomVariable
+            Infection feedback pmf.
+        infections_mean_varname : str, optional
+            Name to be assigned to the deterministic variable in the model.
+            Defaults to "latent_infections".
+
+        Returns
+        -------
+        None
+        """
+
+        self.validate(infection_feedback_strength, infection_feedback_pmf)
+
+        self.infection_feedback_strength = infection_feedback_strength
+        self.infection_feedback_pmf = infection_feedback_pmf
+        self.infections_mean_varname = infections_mean_varname
+
+        return None
+
+    @staticmethod
+    def validate(
+        inf_feedback_strength: any,
+        inf_feedback_pmf: any,
+    ) -> None:  # numpydoc ignore=GL08
+        """
+        Validates the input parameters.
+
+        Parameters
+        ----------
+        inf_feedback_strength : RandomVariable
+            Infection feedback strength.
+        inf_feedback_pmf : RandomVariable
+            Infection feedback pmf.
+
+        Returns
+        -------
+        None
+        """
+        _assert_sample_and_rtype(inf_feedback_strength)
+        _assert_sample_and_rtype(inf_feedback_pmf)
+
+        return None
+
+    def sample(
+        self,
+        Rt: ArrayLike,
+        I0: ArrayLike,
+        gen_int: ArrayLike,
+        **kwargs,
+    ) -> InfectionsRtFeedbackSample:
+        """
+        Samples infections given Rt, initial infections, and generation
+        interval.
+
+        Parameters
+        ----------
+        Rt : ArrayLike
+            Reproduction number.
+        I0 : ArrayLike
+            Initial infections, as an array
+            of the same length as the generation
+            interval PMF.
+        gen_int : ArrayLike
+            Generation interval PMF.
+        **kwargs : dict, optional
+            Additional keyword arguments passed through to internal
+            sample calls, should there be any.
+
+        Returns
+        -------
+        InfectionsWithFeedback
+            Named tuple with "infections".
+        """
+        if I0.size < gen_int.size:
+            raise ValueError(
+                "Initial infections must be at least as long as the "
+                f"generation interval. Got {I0.size} initial infections "
+                f"and {gen_int.size} generation interval."
+            )
+
+        gen_int_rev = jnp.flip(gen_int)
+
+        I0 = I0[: gen_int_rev.size]
```

Comment:
```suggestion
        I0 = I0[-gen_int_rev.size:]
```

Reply:
To me, is clearer the original version, but I have no strong feelings about it.